odoo/orm/models.py

```diff
@@ -3355,33 +3367,108 @@ def _add_sql_constraints(self):
         in _sql_constraints.
 
         """
-        cr = self._cr
-        foreign_key_re = re.compile(r'\s*foreign\s+key\b.*', re.I)
-
-        for (key, definition, message) in self._sql_constraints:
-            conname = '%s_%s' % (self._table, key)
-            if len(conname) > 63:
-                hashed_conname = sql.make_identifier(conname)
-                current_definition = sql.constraint_definition(cr, self._table, hashed_conname)
-                if not current_definition:
-                    _logger.info("Constraint name %r has more than 63 characters, internal PG identifier is %r", conname, hashed_conname)
-                conname = hashed_conname
-            else:
-                current_definition = sql.constraint_definition(cr, self._table, conname)
-            if current_definition == definition:
-                continue
+        for cons in self._sql_constraints:
+            cons.sync_database_object(self)
 
-            if current_definition:
-                # constraint exists but its definition may have changed
-                sql.drop_constraint(cr, self._table, conname)
+    @api.model
+    def __get_columns_from_sql_diagnostics(self, diagnostics, *, check_registry=False) -> list[str]:
```

Comment:
This was originally a function (instead of a method).  Is it a good idea to turn it into a method?

Reply:
I will turn it back into a function.